### ISSUE TITLE
Runtime exception on `copy`

### ISSUE THREAD

**Issue by Technifocal** [gilbertchen/duplicacy#79]
Hi, I'm attempting to copy some backups from `B2` to `GCD`, `GCD` is currently empty, `B2` has ~35GB of data over ~30 revisions.

I ran the following code to add GCD:

`duplicacy add gcd ${SAME-SNAPSHOT-ID-AS-B2} gcd://path/to/backup`

Which ran fine and created the following directory structure on GDrive:

![image](https://user-images.githubusercontent.com/20205745/27254811-79a7254a-5388-11e7-86d1-da16b10f3564.png)

Unfortunately, however, when running:

`duplicacy copy -to gcd`

I receive:
```$ duplicacy copy -to gcd
Source storage set to b2://duplicacy-container
Destination storage set to gcd://duplicacy/path
runtime error: invalid memory address or nil pointer dereference
goroutine 118 [running]:
runtime/debug.Stack(0x41, 0x0, 0x0)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x80
runtime/debug.PrintStack()
        /usr/local/go/src/runtime/debug/stack.go:16 +0x29
github.com/gilbertchen/duplicacy/src.CatchLogException()
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_log.go:173 +0x1a8
panic(0xa71940, 0xe5ab50)
        /usr/local/go/src/runtime/panic.go:489 +0x2dd
github.com/gilbertchen/duplicacy/src.(*Chunk).GetHash(0xc0421561e0, 0x0, 0xc042142140)
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_chunk.go:136 +0x80
github.com/gilbertchen/duplicacy/src.(*ChunkUploader).Upload(0xc042184af0, 0x0, 0xc0421561e0, 0x1, 0xc04203b938)
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_chunkuploader.go:126 +0x236
github.com/gilbertchen/duplicacy/src.(*ChunkUploader).Start.func1(0xc042184af0, 0x0)
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_chunkuploader.go:60 +0x119
created by github.com/gilbertchen/duplicacy/src.(*ChunkUploader).Start
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_chunkuploader.go:65 +0x5b
```

I am running this on Windows 10 Pro x64 (Using the official Windows build bundled with the GUI version 2.0.2 installer).

Here is the log with `-verbose -debug`:

```
$ duplicacy -verbose -debug copy -to gcd
Source storage set to b2://duplicacy-container
Reading the environment variable DUPLICACY_B2_ID
Reading the environment variable DUPLICACY_B2_KEY
Reading the environment variable DUPLICACY_PASSWORD
Compression level: 100
Average chunk size: 4194304
Maximum chunk size: 16777216
Minimum chunk size: 1048576
Chunk seed: 5bdd50ba501ddeca032efad4f9cd8cb3a3326a3c1d85f13500899360e7110337
Destination storage set to gcd://duplicacy/path
Reading the environment variable DUPLICACY_GCD_GCD_TOKEN
Reading the environment variable DUPLICACY_GCD_PASSWORD
Compression level: 100
Average chunk size: 4194304
Maximum chunk size: 16777216
Minimum chunk size: 1048576
Chunk seed: 5bdd50ba501ddeca032efad4f9cd8cb3a3326a3c1d85f13500899360e7110337
Listing all snapshot ids
Listing revisions for snapshot DESKTOP-D
Loaded file snapshots/DESKTOP-D/1 from the snapshot cache
Loaded file snapshots/DESKTOP-D/6 from the snapshot cache
Loaded file snapshots/DESKTOP-D/7 from the snapshot cache
Loaded file snapshots/DESKTOP-D/8 from the snapshot cache
Loaded file snapshots/DESKTOP-D/9 from the snapshot cache
Loaded file snapshots/DESKTOP-D/10 from the snapshot cache
Loaded file snapshots/DESKTOP-D/11 from the snapshot cache
Loaded file snapshots/DESKTOP-D/12 from the snapshot cache
Loaded file snapshots/DESKTOP-D/13 from the snapshot cache
Loaded file snapshots/DESKTOP-D/14 from the snapshot cache
Loaded file snapshots/DESKTOP-D/15 from the snapshot cache
Loaded file snapshots/DESKTOP-D/16 from the snapshot cache
Loaded file snapshots/DESKTOP-D/17 from the snapshot cache
Loaded file snapshots/DESKTOP-D/18 from the snapshot cache
Loaded file snapshots/DESKTOP-D/19 from the snapshot cache
Loaded file snapshots/DESKTOP-D/20 from the snapshot cache
Loaded file snapshots/DESKTOP-D/21 from the snapshot cache
Loaded file snapshots/DESKTOP-D/22 from the snapshot cache
Loaded file snapshots/DESKTOP-D/23 from the snapshot cache
Loaded file snapshots/DESKTOP-D/24 from the snapshot cache
Loaded file snapshots/DESKTOP-D/25 from the snapshot cache
Loaded file snapshots/DESKTOP-D/26 from the snapshot cache
Loaded file snapshots/DESKTOP-D/27 from the snapshot cache
Loaded file snapshots/DESKTOP-D/28 from the snapshot cache
Loaded file snapshots/DESKTOP-D/29 from the snapshot cache
Loaded file snapshots/DESKTOP-D/30 from the snapshot cache
Loaded file snapshots/DESKTOP-D/31 from the snapshot cache
Loaded file snapshots/DESKTOP-D/32 from the snapshot cache
Copying snapshot DESKTOP-D at revision 1
Chunk db847e3e24e038cf9f66c2eee5eeeb5d53e6e531dca503670da2ca142de8f06d has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 6
Fetching chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2
Chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 7
Fetching chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2
Chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 8
Fetching chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2
Chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 9
Fetching chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2
Chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 10
Fetching chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2
Chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 11
Fetching chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2
Chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 12
Fetching chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2
Chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 13
Fetching chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2
Chunk 3160b59c4ba36e4932ad9360a42e1ff57f1618f09dce7d14d3c476a17f91dbd2 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 14
Fetching chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da
Chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 15
Fetching chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da
Chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 16
Fetching chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da
Chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 17
Fetching chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da
Chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 18
Fetching chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da
Chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 19
Fetching chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da
Chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 20
Fetching chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da
Chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 21
Fetching chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da
Chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 22
Fetching chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da
Chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 23
Fetching chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da
Chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 24
Fetching chunk 294a49575019fcd80e2cc6b9195e0c9e5a4a80413aa3c3c45a0593e18bf759c9
Chunk 294a49575019fcd80e2cc6b9195e0c9e5a4a80413aa3c3c45a0593e18bf759c9 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 25
Fetching chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da
Chunk 01b9850b537d5bfaeb1eac6b9031b3219faad5fa0de2a17a4b137f74420791da has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 26
Fetching chunk c0158600fbc9a6b597338fbeef6e767cd15427f0e91c191dd566303ebdff066b
Chunk c0158600fbc9a6b597338fbeef6e767cd15427f0e91c191dd566303ebdff066b has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 27
Fetching chunk 4bfd12c5b63ceb9fdb8a1325a0dff6d1b373a7b7b068effa01b6ae5735816501
Chunk 4bfd12c5b63ceb9fdb8a1325a0dff6d1b373a7b7b068effa01b6ae5735816501 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 28
Fetching chunk 4bfd12c5b63ceb9fdb8a1325a0dff6d1b373a7b7b068effa01b6ae5735816501
Chunk 4bfd12c5b63ceb9fdb8a1325a0dff6d1b373a7b7b068effa01b6ae5735816501 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 29
Fetching chunk 4bfd12c5b63ceb9fdb8a1325a0dff6d1b373a7b7b068effa01b6ae5735816501
Chunk 4bfd12c5b63ceb9fdb8a1325a0dff6d1b373a7b7b068effa01b6ae5735816501 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 30
Fetching chunk 4bfd12c5b63ceb9fdb8a1325a0dff6d1b373a7b7b068effa01b6ae5735816501
Chunk 4bfd12c5b63ceb9fdb8a1325a0dff6d1b373a7b7b068effa01b6ae5735816501 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 31
Fetching chunk 4bfd12c5b63ceb9fdb8a1325a0dff6d1b373a7b7b068effa01b6ae5735816501
Chunk 4bfd12c5b63ceb9fdb8a1325a0dff6d1b373a7b7b068effa01b6ae5735816501 has been loaded from the snapshot cache
Copying snapshot DESKTOP-D at revision 32
Fetching chunk 4bfd12c5b63ceb9fdb8a1325a0dff6d1b373a7b7b068effa01b6ae5735816501
Chunk 4bfd12c5b63ceb9fdb8a1325a0dff6d1b373a7b7b068effa01b6ae5735816501 has been loaded from the snapshot cache
Copying chunk db8c116c5a010a1f84701731984b539d1a98cef5f5a3196e06b302fa2f673b8f to 8715c60c4a74a2a0c24ac67cf4c37554fcba8f8b2962c4913892990c666e4252
Chunk db8c116c5a010a1f84701731984b539d1a98cef5f5a3196e06b302fa2f673b8f has been downloaded
Copying chunk 4190e23b1a574db6139c9c895fac01b49445289cb64c5a2df36ada0197193fa6 to 39dc0b516948f29cf1792beb6fd76d436ca798a33996d16b180cfc3c49f5225a
Fetching chunk 4190e23b1a574db6139c9c895fac01b49445289cb64c5a2df36ada0197193fa6
runtime error: invalid memory address or nil pointer dereference
goroutine 140 [running]:
runtime/debug.Stack(0x41, 0x0, 0x0)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x80
runtime/debug.PrintStack()
        /usr/local/go/src/runtime/debug/stack.go:16 +0x29
github.com/gilbertchen/duplicacy/src.CatchLogException()
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_log.go:173 +0x1a8
panic(0xa71940, 0xe5ab50)
        /usr/local/go/src/runtime/panic.go:489 +0x2dd
github.com/gilbertchen/duplicacy/src.(*Chunk).GetHash(0xc0422c4a50, 0x0, 0xc042248200)
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_chunk.go:136 +0x80
github.com/gilbertchen/duplicacy/src.(*ChunkUploader).Upload(0xc0422c49b0, 0x0, 0xc0422c4a50, 0x1, 0x0)
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_chunkuploader.go:126 +0x236
github.com/gilbertchen/duplicacy/src.(*ChunkUploader).Start.func1(0xc0422c49b0, 0x0)
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_chunkuploader.go:60 +0x119
created by github.com/gilbertchen/duplicacy/src.(*ChunkUploader).Start
        /Users/chgang/zincbox/go/src/github.com/gilbertchen/duplicacy/src/duplicacy_chunkuploader.go:65 +0x5b
```

**Comment by gilbertchen** [comment]
This is the same as #76 and has been fixed in the master branch.  Let me know if you need a binary built from the latest source.

**Comment by gilbertchen** [comment]
The fix is in release 2.0.3
